### PR TITLE
Fetch index and component

### PR DESCRIPTION
**Pull request by july2993**
Fetch index and component

### CODE REVIEW

File: pkg/repository/v1_repository.go [hunk 1]
```diff
@@ -338,3 +338,39 @@ func (r *V1Repository) fetchManifest(url string, role v1manifest.ValidManifest,
 	defer reader.Close()
 	return v1manifest.ReadManifest(reader, role, root)
 }
+
+func (r *V1Repository) ensureManifests() error {
+	// TODO
+	return nil
```
Comment (lonng) [inline pkg/repository/v1_repository.go:344]: Will this TODO will be addressed in this PR?

Comment (july2993) [reply pkg/repository/v1_repository.go:344]: supporst #230 will do this.

**Comment by july2993** [comment]
export one to get Index and Component is request by @lucklove

**Comment by nrc** [comment]
> export one to get Index and Component is request by @lucklove

@lucklove why do you want these functions?

**Comment by lucklove** [comment]
> > export one to get Index and Component is request by @lucklove
> 
> @lucklove why do you want these functions?

because `tiup list` and `tiup list <comp>` need newest index and comp manifest

**Pull request merged**